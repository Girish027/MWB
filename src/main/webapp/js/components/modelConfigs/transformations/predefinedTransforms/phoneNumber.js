import Constants from 'constants/Constants';

const phoneNumber = () => {
  const regex = {
    type: Constants.TRANSFORMATION_TYPES.WORDCLASS_SUBST_REGEX,
    mappings: {},

  };
  regex.mappings = {
    '/\\(?([0-9]{3})?[-.\\s)]*[0-9]{3}[-.\\s]*[0-9]{4}\\b/': '_class_phone_number',
    '/\\b[0-9]{5}[\\s\\.-]?[0-9]{5}\\b/': '_class_phone_number',
  };
  return regex;
};

export default phoneNumber;
