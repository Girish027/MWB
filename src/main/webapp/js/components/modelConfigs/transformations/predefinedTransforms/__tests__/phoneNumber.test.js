import phoneNumber from 'components/modelConfigs/transformations/predefinedTransforms/phoneNumber';

describe('phoneNumber', () => {
  const regex = phoneNumber();

  test('type is equal to wordclass-subst-regex', () => {
    expect(regex.type).toEqual('wordclass-subst-regex');
  });

  test('mapping contains more than one key', () => {
    const keys = Object.keys(regex.mappings);
    expect(keys.length).toBeGreaterThan(1);
  });

  test('snapshot predefined transformation', () => {
    expect(regex.mappings).toMatchSnapshot();
  });
});
