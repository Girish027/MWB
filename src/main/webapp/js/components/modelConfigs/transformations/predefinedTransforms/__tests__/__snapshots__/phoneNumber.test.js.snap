// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phoneNumber snapshot predefined transformation 1`] = `
Object {
  "/\\\\(?([0-9]{3})?[-.\\\\s)]*[0-9]{3}[-.\\\\s]*[0-9]{4}\\\\b/": "_class_phone_number",
  "/\\\\b[0-9]{5}[\\\\s\\\\.-]?[0-9]{5}\\\\b/": "_class_phone_number",
}
`;
